torch/optim/adam.py

```diff
@@ -405,3 +487,85 @@ def _multi_tensor_adam(params: List[Tensor],
             denom = torch._foreach_add(exp_avg_sq_sqrt, eps)
 
         torch._foreach_addcdiv_(params, exp_avgs, denom, step_size)
+
+
+# TODO(crcrpar): Move this to another place when adding another fused optimizer.
+# TODO(crcrpar): Make this generic when there's more fused optimizers.
+# TODO(crcrpar): Think of rewriting this in C++.
+@torch.no_grad()
+def _group_params_by_device_and_dtype(
+    params: List[Tensor],
```

Comment:
you probably can make this function more generic by using *args instead of named arguments, and just iterating over arguments list inside.